provider/provider.go

```diff
@@ -170,9 +170,20 @@ func (p *Provider) FetchResources(ctx context.Context, request *cqproto.FetchRes
 	defer conn.Close()
 
 	// limiter used to limit the amount of resources fetched concurrently
-	var limiter *semaphore.Weighted
+	var goroutinesSem *semaphore.Weighted
+	if request.MaxGoroutines == 0 {
+		// TODO: I don't like this conversion to int64
+		goroutinesSem = semaphore.NewWeighted(int64(helpers.GetMaxGoRoutines()))
+	} else {
+		// TODO: I don't like this conversion to int64
+		goroutinesSem = semaphore.NewWeighted(int64(request.MaxGoroutines))
+	}
```

Comment:
would add another cases which is -1 i.e no disable would then make this a switch case.

Reply:
it's uint64 so it can't be `-1` also if someone specify `0` it will mean it can't run so it's not a valid value as the minimum value is `1`